app/src/main/java/org/p2p/wallet/swap/ui/jupiter/info/SwapInfoBottomSheet.kt

```diff
@@ -0,0 +1,210 @@
+
+    private val managerHolder: SwapStateManagerHolder by inject()
+    private val mapper: SwapInfoMapper by inject()
+    private val interactor: SwapTokensInteractor by inject {
+        parametersOf(stateManagerKey)
+    }
+    private val coroutineDispatchers: CoroutineDispatchers by inject()
+    private val stateManager: SwapStateManager
+        get() = managerHolder.get(stateManagerKey)
+
+    private val adapter = SwapSettingsAdapter(
+        baseCellDelegate(),
+        infoBlockCellDelegate(),
+        swapInfoBannerDelegate(),
+    )
+
+    override fun getTheme(): Int = R.style.WalletTheme_BottomSheet_Rounded
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.dialog_swap_info, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.recyclerViewRoutes.attachAdapter(adapter)
+        binding.buttonDone.setOnClickListener { dismiss() }
+        when (swapInfoType) {
+            SwapInfoType.MINIMUM_RECEIVED -> {
+                adapter.items = mapper.mapMinimumReceived()
+                binding.buttonDone.setText(R.string.swap_info_details_minimum_received_fee_done)
+            }
+            SwapInfoType.NETWORK_FEE -> {
+                adapter.items = mapper.mapNetworkFee()
+                binding.buttonDone.setText(R.string.swap_info_details_network_fee_done)
+            }
+            SwapInfoType.ACCOUNT_FEE -> {
+                adapter.items = mapper.mapAccountFee()
+                binding.buttonDone.setText(R.string.swap_info_details_account_fee_done)
+            }
+            SwapInfoType.LIQUIDITY_FEE -> {
+                binding.buttonDone.setText(R.string.swap_info_details_liquidity_fee_done)
+                observeFeatureState()
+            }
+        }
+    }
+
+    private fun observeFeatureState() {
+        val allTokens = flow { emit(interactor.getAllTokens()) }
+        stateManager.observe().combine(allTokens) { state, tokens ->
+            state to tokens
+        }
+            .flatMapLatest {
+                val (state, tokens) = it
+                handleFeatureState(state, tokens)
+            }
+            .flowOn(coroutineDispatchers.io)
+            .onEach { adapter.items = it }
+            .flowWithLifecycle(lifecycle)
+            .launchIn(lifecycleScope)
+    }
+
+    private suspend fun handleFeatureState(state: SwapState, tokens: List<SwapTokenModel>): Flow<List<AnyCellItem>> {
+        return when (state) {
+            SwapState.InitialLoading,
+            is SwapState.LoadingRoutes,
+            is SwapState.TokenAZero -> flowOf(mapper.mapEmptyLiquidityFee())
+            is SwapState.SwapException -> handleFeatureState(state.previousFeatureState, tokens)
+            is SwapState.SwapLoaded -> {
```

Comment:
looks like we need a presenter here..

Reply:
refactoring later if add some local properties